subprojects/core/src/main/java/org/gradle/api/internal/artifacts/dsl/dependencies/ConfigurationDependencyHandlerExtensionModule.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.artifacts.dsl.dependencies;
+
+import groovy.lang.Closure;
+import groovy.transform.stc.ClosureParams;
+import groovy.transform.stc.SecondParam;
+import groovy.transform.stc.SimpleType;
+import org.gradle.api.Project;
+import org.gradle.api.artifacts.Dependency;
+import org.gradle.api.artifacts.MinimalExternalModuleDependency;
+import org.gradle.api.artifacts.dsl.DependencyAdder;
+import org.gradle.api.artifacts.dsl.DependencyFactory;
+import org.gradle.api.file.FileCollection;
+import org.gradle.api.provider.Provider;
+import org.gradle.api.provider.ProviderConvertible;
+import org.gradle.util.internal.ConfigureUtil;
+
+import java.util.Map;
+
+public class ConfigurationDependencyHandlerExtensionModule {
+    /**
+     * Add a dependency.
+     *
+     * @param dependencyNotation dependency to add
+     * @see DependencyFactory#createFromCharSequence(CharSequence) Valid dependency notation for this method
+     */
+    public static void call(DependencyAdder self, CharSequence dependencyNotation) {
+        self.add(dependencyNotation);
+    }
+
+    /**
+     * Add a dependency.
+     *
+     * @param dependencyNotation dependency to add
+     * @param configuration an action to configure the dependency
+     * @see DependencyFactory#createFromCharSequence(CharSequence) Valid dependency notation for this method
+     */
+    public static void call(DependencyAdder self, CharSequence dependencyNotation, @ClosureParams(value = SimpleType.class, options = "org.gradle.api.artifacts.ExternalModuleDependency") Closure<?> configuration) {
```

Comment:
Actions don't work properly in Groovy, they do not get a receiver like in Kotlin DSL. I would kinda like to fix that, but I'm not sure how to, I think it would be a new feature in Groovy.